### PR TITLE
Allow user to create collection of enum with doctrine

### PR DESCRIPTION
**Pull request by benji07**
- [x] Documentation
- [x] Mark legacy format of configuration as deprecated

### CODE REVIEW

File: src/Bridge/Symfony/Bundle/DependencyInjection/Configuration.php [hunk 1]
```diff
@@ -49,24 +49,53 @@ private function addDoctrineSection(ArrayNodeDefinition $rootNode)
...
+                            } else {
+                                $newValues = $values;
+                            }
+
+                            return $newValues;
+                        })
+                    ->end()
                     ->validate()
                         ->ifTrue(static function (array $v): bool {return self::hasNonEnumKeys($v); })
                         ->then(static function (array $v) { self::throwsNonEnumKeysException($v); })
```
Comment (ogizanagi) [inline src/Bridge/Symfony/Bundle/DependencyInjection/Configuration.php:81]:
```suggestion
                        ->ifTrue(static function (array $v): bool {return self::hasNonEnumClasses($v); })
                        ->then(static function (array $v) { self::throwsNonEnumClassesException($v); })
```

Comment (ogizanagi) [reply src/Bridge/Symfony/Bundle/DependencyInjection/Configuration.php:81]: Can't we add this validation on the `class` node directly now it's not the keys anymore?

File: src/Bridge/Symfony/Bundle/DependencyInjection/Configuration.php [hunk 2]
```diff
@@ -55,7 +55,7 @@ private function addDoctrineSection(ArrayNodeDefinition $rootNode)
                             foreach ($values as $name => $config) {
                                 if (class_exists($name)) {
                                     $legacyFormat = true;
-                                    // need to flip
+                                    @trigger_error('The format has change at path elao_enum.doctrine.types, you must index your configuration array by name and not by class fqcn', E_USER_DEPRECATED);
```
Comment (ogizanagi) [inline src/Bridge/Symfony/Bundle/DependencyInjection/Configuration.php:58]:
```suggestion
                                    @trigger_error('Using enum FQCN as keys at path "elao_enum.doctrine.types" is deprecated. provide the name as keys and add the "class" option for each entry instead.', E_USER_DEPRECATED);
```

File: tests/Fixtures/Bridge/Symfony/Bundle/DependencyInjection/ElaoEnumExtension/xml/doctrine_types_legacy.xml [hunk 3]
```diff
@@ -0,0 +1,15 @@
+<?xml version="1.0" ?>
+
+<container xmlns="http://symfony.com/schema/dic/services"
+           xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+           xmlns:elao-enum="http://elao.com/schema/dic/elao_enum"
+           xsi:schemaLocation="http://elao.com/schema/dic/elao_enum http://elao.com/schema/dic/elao_enum/elao_enum.xsd">
+
+    <elao-enum:config>
```
Comment (ogizanagi) [inline tests/Fixtures/Bridge/Symfony/Bundle/DependencyInjection/ElaoEnumExtension/xml/doctrine_types_legacy.xml:8]: No changes for the XML format, so you can simply skip it in the test case

Comment (benji07) [reply tests/Fixtures/Bridge/Symfony/Bundle/DependencyInjection/ElaoEnumExtension/xml/doctrine_types_legacy.xml]: yes, i just added the test in the wrong test case

**Comment by ogizanagi** [comment]
Thanks @benji07.

**Pull request closed without merging**